### PR TITLE
fix missing nopython keyword in cuda random module

### PR DESCRIPTION
**Pull request by esc**
Fixes: #8951

### CODE REVIEW

**Comment by gmarkall** [comment]
gpuci run tests

File: numba/cuda/random.py [hunk 1]
```diff
@@ -34,15 +34,16 @@
 xoroshiro128p_type = from_dtype(xoroshiro128p_dtype)
 
 # When cudasim is enabled, Fake CUDA arrays are passed to some of the
-# @jit-decorated functions. This causes a fallback to object mode and the
-# emission of a warning at present - this is visible in the test suite with
-# cudasim, but also fallback-by-default will not be permitted in future. In
-# order to avoid the warning / future error, we explicitly specify that
+# @jit-decorated functions. This required the a fallback to object mode. With
```
Comment (gmarkall) [inline numba/cuda/random.py:37]: "the a" - should this be:
```suggestion
# @jit-decorated functions. This required fallback to object mode. With
```

File: numba/cuda/random.py [hunk 2]
```diff
@@ -34,15 +34,16 @@
...
+# Numba 0.59.0 this fallback must be explicitly enabled.
```
Comment (gmarkall) [inline numba/cuda/random.py:38]: I think there is no fallback in the future:

```suggestion
# Numba 0.59.0 object mode must be explicitly enabled.
```

**Review by gmarkall — CHANGES_REQUESTED** [review]
The code changes look good to me in their present form (assuming gpuCI passes) - just a couple of comments on the explanatory comments.

**Comment by esc** [comment]
@gmarkall thank you for the review, I have addressed the review comments accordingly and pushed two new commits.

**Comment by gmarkall** [comment]
gpuci run tests

**Review by gmarkall — APPROVED** [review]
Many thanks for the fix - this looks good to me!

**Pull request merged**